pkg/bootstrap/stage.go

```diff
@@ -166,7 +166,19 @@ func extractFromDir(dir, prefix string, img v1.Image, imgName string) error {
 	if err := extract(imgName, tempDir, prefix, r); err != nil {
 		return err
 	}
-	return os.Rename(tempDir, dir)
+	if err := os.Rename(tempDir, dir); err != nil {
+		files, err := ioutil.ReadDir(tempDir)
+		if err != nil {
+			return err
+		}
+		for _, file := range files {
+			if err := os.Rename(filepath.Join(tempDir, file.Name()), filepath.Join(dir, file.Name())); err != nil {
+				os.Remove(filepath.Join(dir, file.Name()))
+				os.Rename(filepath.Join(tempDir, file.Name()), filepath.Join(dir, file.Name()))
+			}
+		}
+	}
+	return nil
```

Comment:
Need to remove the temp directory, since we didn't rename it.

Reply:
https://github.com/rancher/rke2/blob/43f82980a947857fddf4f8fec2cf1ee4aec288b9/pkg/bootstrap/stage.go#L161